### PR TITLE
Added support for Cisco 8000 virtual sonic DUT

### PR DESCRIPTION
**Pull request by rskorka**
### Description of PR 

Summary:
Add playbooks which allow to use T0/T1 virtual testbeds with Cisco 8000e emulator (functional emulator of Cisco 8000 Series routers).  

Note: Cisco provides Cisco 8000e emulator to its customers.

### Type of change

- [ ] Bug fix
- [ x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012

#### How did you do it?
Add start/stop playbooks for new DUT type: 8000e.
The playbook starts the emulator inside a docker container (for ease of deployment) and then vm-topology module can connect it to the topology.

#### How did you verify/test it?

./testbed-cli.sh -m veos_vtb -n 4 -k vsonic start-vms server_1 password.txt
./testbed-cli.sh -k vsonic -t vtestbed.yaml -m veos_vtb add-topo 8000e-t0 password.txt
./testbed-cli.sh -t vtestbed.yaml -m veos_vtb deploy-mg 8000e-t0 veos_vtb password.txt
cd ../tests
./run_tests.sh -n 8000e-t0 -d vlab-8k-01 -c bgp/test_bgp_fact.py -f ../ansible/vtestbed.yaml -i ../ansible/veos_vtb -e --disable_loganalyzer

bgp/test_bgp_fact.py::test_bgp_facts[vlab-8k-01-None] PASSED  

#### Any platform specific information?
While the 8000e emulator supports many Cisco 8000 platforms, the new 8000e playbooks support Cisco-8102-C64 platform specifically. Other 8000e platforms will be tested and enabled in the future.

### CODE REVIEW

File: ansible/roles/vm_set/tasks/start_8000e_sonic.yml [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+  when: hwsku == "Cisco-8102-C64"
+
+- name: capture all host interfaces
+  shell: ifconfig -a |egrep "^[0-9a-zA-Z]" | awk '{ print $1 }'
+  register: host_interfaces
+
+
+- name: capture host "up" interfaces
+  shell: ifconfig |egrep "^[0-9a-zA-Z]" | awk '{ print $1 }'
+  register: host_up_interfaces
```
Comment (wangxin) [inline ansible/roles/vm_set/tasks/start_8000e_sonic.yml:69]: Same `become: yes` concern.

File: ansible/roles/vm_set/tasks/start_8000e_sonic.yml [hunk 2]
```diff
@@ -0,0 +1,116 @@
+- set_fact:
+        docker_image:  8000e-sonic
+        mgmt_ip_address: " {{ hostvars[dut_name]['ansible_host'] }}"
+        mgmt_gw: "{{ vm_mgmt_gw | default(mgmt_gw) }}"
+        sonic_image: "{{ home_path }}/8000e/images/sonic-cisco-8000.bin"
```
[the remark this answers is not in the document]
Comment (rskorka) [reply ansible/roles/vm_set/tasks/start_8000e_sonic.yml:5]: Yes, Cisco customers can contact account team for image access. I will add installation instructions in the README file.

File: ansible/vtestbed.yaml [hunk 3]
```diff
@@ -184,3 +184,33 @@
...
   auto_recover: 'False'
   comment: Dual-TOR-Mixed testbed
+
+- conf-name: 8000e-t0
+  group-name: vms6-1
+  topo: t0
+  ptf_image_name: docker-ptf
+  ptf: ptf-01
+  ptf_ip: 10.250.0.102/24
+  ptf_ipv6: fec0::ffff:afa:2/64
```
[the remark this answers is not in the document]
Comment (rskorka) [reply ansible/vtestbed.yaml:194]: Is there a specific IP subnet range I should be using after I assign a new PTF?

File: ansible/vtestbed.yaml [hunk 4]
```diff
@@ -184,3 +184,33 @@
...
+  auto_recover: 'False'
+  comment: Tests 8000e sonic device
+
+- conf-name: 8000e-t1
+  group-name: vms6-1
+  topo: t1
+  ptf_image_name: docker-ptf
+  ptf: ptf-01
+  ptf_ip: 10.250.0.102/24
+  ptf_ipv6: fec0::ffff:afa:2/64
```
[the remark this answers is not in the document]
Comment (rskorka) [reply ansible/vtestbed.yaml]: Fixed

File: ansible/vtestbed.csv [hunk 5]
```diff
@@ -11,3 +11,5 @@ vms-kvm-t2,vms6-4,t2-vs,docker-ptf,ptf-04,10.250.0.109/24,fec0::ffff:afa:9/64,se
 vms-kvm-t0-3,vms6-6,t0,docker-ptf,ptf-06,10.250.0.116/24,fec0::ffff:afb:2/64,server_1,VM0132,[vlab-09],veos_vtb,False,Tests virtual switch vm
 vms-kvm-t0-4,vms6-7,t0,docker-ptf,ptf-07,10.250.0.118/24,fec0::ffff:afb:4/64,server_1,VM0136,[vlab-10],veos_vtb,False,Tests virtual switch vm
 vms-kvm-dual-mixed,vms6-8,dualtor-mixed,docker-ptf,ptf-08,10.250.0.109/24,fec0::ffff:afa:9/64,server_1,VM0140,[vlab-11;vlab-12],veos_vtb,False,Dual-TOR-Mixed testbed
+8000e-t0,vms6-1,t0,docker-ptf,ptf-01,10.250.0.102/24,fec0::ffff:afa:2/64,server_1,VM0100,[vlab-8k-01],veos_vtb,False,Tests 8000e sonic device
+8000e-t1,vms6-1,t1,docker-ptf,ptf-01,10.250.0.102/24,fec0::ffff:afa:2/64,server_1,VM0100,[vlab-8k-01],veos_vtb,False,Tests 8000e sonic device
```
[the remark this answers is not in the document]
Comment (rskorka) [reply ansible/vtestbed.csv]: I removed 8000e references from vtestbed.csv.

**Pull request merged**